<?xml version="1.0"?>
<!--
  自动驾驶仿真与规划系统启动文件
  
  启动以下节点:
  - 仿真节点 (simulation_node): 生成地图和规划请求
  - 规划节点 (path_planner_node): 执行A*和Hybrid A*路径规划
  - RViz可视化工具: 显示模拟结果
-->
<launch>
  <!-- 启动参数 -->
  <arg name="use_rviz" default="true" description="是否启动RViz可视化"/>
  <arg name="planner_type" default="hybrid_astar" description="规划器类型: astar或hybrid_astar"/>
  
  <!-- 启动仿真节点 -->
  <node pkg="auto_simulation" exec="simulation_node" name="simulation_node">
    <param name="map_update_interval" value="5.0"/>
    <param name="planning_interval" value="10.0"/>
  </node>

  <!-- 启动规划节点 -->
  <node pkg="auto_planning" exec="path_planner_node" name="path_planner_node">
    <param name="default_planner" value="$(var planner_type)"/>
  </node>
  
  <!-- 启动感知节点 -->
  <node pkg="auto_perception" exec="perception_node" name="perception_node">
  </node>
  
  <!-- 启动控制节点 -->
  <node pkg="auto_control" exec="controller_node" name="controller_node">
    <param name="lookahead_distance" value="3.0"/>
    <param name="max_velocity" value="5.0"/>
  </node>
  
  <!-- 启动决策可视化节点 -->
  <node pkg="auto_planning" exec="decision_visualization_node" name="decision_visualization_node">
  </node>
  
  <!-- 启动MQTT桥接节点 -->
  <node pkg="auto_simulation" exec="mqtt_bridge_node" name="mqtt_bridge_node">
    <param name="mqtt_host" value="localhost"/>
    <param name="mqtt_port" value="1883"/>
    <param name="mqtt_client_id" value="auto_driving_bridge"/>
  </node>
  
  <!-- 如果需要，启动RViz可视化 -->
  <group if="$(var use_rviz)">
    <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share auto_simulation)/config/auto_driving.rviz"/>
  </group>
</launch>
